CONTRIBUTING.md

```diff
@@ -0,0 +1,176 @@
+```
+
+Running this command will create a `Vagrantfile` in your working directory which
+is used to configure your vagrant box.
+
+Before we start our Vagrant box, we will need to increase the amount of RAM available
+to the system. 
+To do this, edit the Vagrantfile and configure your provider settings to include
+the following:
+
+```rb
+    config.vm.provider "libvirt" do |v|
+        # provides 3GB of memory
+        v.memory = 3072
+        # for parallelization
+        v.cpus = 2
+    end
+```
+
+The value of `config.vm.provider` depends on the provider you selected when you
+ran `vagrant add` earlier. For example, if you selected virtualbox then the first
+line should be: `config.vm.provider "virtualbox" do |v|`
+
+
+Now we can start the VM:
+
+```
+vagrant up
+```
+
+Once the VM is up, connect to it:
+
+```
+vagrant ssh
+```
+
+### (Extra Optional) Connecting VSCode to Vagrant 
+
+If you're using VSCode, you can connect to your vagrant box with a few extra steps.
+
+First we need to ask Vagrant for an SSH config file. From your host machine, run:
+
+```
+vagrant ssh-config > ssh-config.conf
+```
+
+*You can edit the `ssh-config.conf` file to change the hostname from `default` to
+`vagrant` to be more easily identifiable, but that's up to you. :)*
+
+Here's an example of my working SSH config file:
+```
+Host default
+  HostName 127.0.0.1
+  User vagrant
+  Port 2222
+  UserKnownHostsFile /dev/null
+  StrictHostKeyChecking no
+  PasswordAuthentication no
+  IdentityFile /path/to/microshift/.vagrant/machines/default/virtualbox/private_key
+  IdentitiesOnly yes
+  LogLevel FATAL
+```
+
+
+Next, you'll want to install the `Remote - SSH` extension from the [VSCode Marketplace](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh)
+
+
+
+With the extension installed, you'll click on the green bottom in the bottom-left
+corner of VSCode to open a dropdown menu for SSH options:
+
+![VSCode Remote Button](./docs/pics/vscode-remote-button.png)
+
+Select the option to open an SSH configuration file:
+![Dropdown Menu](./docs/pics/remote-ssh-dropdown.png)
+
+Next you'll want to navigate to the "Remote Explorer" tab on the left-hand side
+of VSCode, then select on the vagrant target (default if you haven't renamed it)
+and click on the button to connect to it in a remote window.
+
+
+*(Credits to Andrés Lopez for this guide: [Connect Visual Studio Code with Vagrant in your local machine
+](https://medium.com/@lopezgand/connect-visual-studio-code-with-vagrant-in-your-local-machine-24903fb4a9de))*
+
+
+## Running Microshift Locally
+
+### Pre-Installation
+
+You will need to install the required binaries:
+
+```sh
+# Fedora/CentOS 
+sudo dnf install \
+    git \
+    make \
+    golang \    
+    glibc-static
+
+# Ubuntu
+sudo apt install \
+    git \
+    build-essential \   # provides Make
+    golang-go \
+    glibc
+```
+
+You will also need to have `kubectl` and `docker` installed, you can follow
```

Comment:
Our target container engine is actually `podman`. `docker` is not tested and so may or may not work.